resources/sdk/clisdkclient/extensions/cmd/profiles/createprofile.go

```diff
@@ -77,35 +104,94 @@ func requestUserInput() config.Configuration {
 		environment = "mypurecloud.com"
 	}
 
-	fmt.Printf("Please provide either an Access Token, Client Credentials (Client ID and Client Secret) or both\n")
-	fmt.Printf("Note: If you provide an Access Token, this will be used over Client Credentials\n")
-	for {
-		fmt.Printf("Access Token: ")
-		fmt.Scanln(&accessToken)
-		fmt.Printf("Client ID: ")
-		fmt.Scanln(&clientID)
-		fmt.Printf("Client Secret: ")
-		fmt.Scanln(&clientSecret)
-		// users should only be allowed to continue if they provide an access token
-		if len(strings.TrimSpace(accessToken)) != 0 && len(strings.TrimSpace(clientID)) == 0 && len(strings.TrimSpace(clientSecret)) == 0 {
-			clientID = ""
-			clientSecret = ""
-			break
-			// or if they provide client credentials
-		} else if len(strings.TrimSpace(accessToken)) == 0 && len(strings.TrimSpace(clientID)) != 0 && len(strings.TrimSpace(clientSecret)) != 0 {
-			accessToken = ""
-			break
-			// or both
-		} else if len(strings.TrimSpace(accessToken)) != 0 && len(strings.TrimSpace(clientID)) != 0 && len(strings.TrimSpace(clientSecret)) != 0 {
+	fmt.Print("Note: If you provide an access token, this will take precedence over any authorization grant type.\n")
+	fmt.Print("Access Token (Optional): ")
+	fmt.Scanln(&accessToken)
+
+	for true {
+		fmt.Print("Select your authorization grant type.\n")
+		fmt.Print("\t0. None\n\t1. Client Credentials\n\t2. Implicit Grant\nGrant Type: ")
+		fmt.Scanln(&grantType)
+
+		if accessToken == "" && grantType == None {
+			fmt.Print("If you have not provided an access token, you must select a grant type.\n")
+			continue
+		}
+		if isValidGrantType(grantType) {
 			break
-			// otherwise
+		}
+	}
+
+	clientID, clientSecret = requestClientCreds(accessToken, grantType)
+
+	if grantType == ImplicitGrant {
+		redirectURI = requestRedirectURI()
+		fmt.Printf("Redirect URI: %s\n", redirectURI)
+		for true {
+			fmt.Print("Would you like to open a secure HTTP connection? [Y/N]: ")
```

Comment:
This should be asked before they are asked for the redirect URI. And `open` should be switched to `use`